File: app/src/main/java/com/lskycity/androidtools/InfoBin.java
package com.lskycity.androidtools;

import android.util.Pair;

/**
 * Created by zhaofliu on 1/29/17.
 *
 * @author zhaofliu
 * @since 1/29/17
 */

public class InfoBin{
    public String name;
    public String value;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        InfoBin bin = (InfoBin) o;

        if (name != null ? !name.equals(bin.name) : bin.name != null) return false;
        return value != null ? value.equals(bin.value) : bin.value == null;

    }

    @Override
    public int hashCode() {
        int result = name != null ? name.hashCode() : 0;
        result = 31 * result + (value != null ? value.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "InfoBin{" +
                "name='" + name + '\'' +
                ", value='" + value + '\'' +
                '}' +"\n";
    }
}

File: app/src/main/java/com/lskycity/androidtools/InfoBin.kt
package com.lskycity.androidtools

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by zhaofliu on 1/29/17.
 *
 * @author zhaofliu
 * @since 1/29/17
 */

data class InfoBin(var name: String?, var value: String?) : Parcelable {

    constructor(parcel: Parcel) : this(parcel.readString(), parcel.readString())
    constructor() : this("", "")

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeString(value)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<InfoBin> {
        override fun createFromParcel(parcel: Parcel): InfoBin {
            return InfoBin(parcel)
        }

        override fun newArray(size: Int): Array<InfoBin?> {
            return arrayOfNulls(size)
        }
    }
}
